src/applications/financial-status-report/components/DependentAges.jsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect, useMemo, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { VaTextInput } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { setData } from 'platform/forms-system/src/js/actions';
+import { DEPENDENT_AGE_LABELS } from '../constants/dependentLabels';
+
+export const isNumber = value => {
+  const pattern = /^[0-9]+$/;
+  return pattern.test(value);
+};
+
+const DependentAges = () => {
+  const dispatch = useDispatch();
+  const formData = useSelector(state => state.form.data);
+  const {
+    questions: { hasDependents } = {},
+    personalData: { dependents = [] } = {},
+  } = formData;
+
+  const [stateDependents, setStateDependents] = useState(dependents);
+  const [errors, setErrors] = useState(
+    Array(stateDependents.length).fill(null),
+  );
+
+  useEffect(
+    () => {
+      const shouldInitializeDependents =
+        !stateDependents.length ||
+        stateDependents.length !== parseInt(hasDependents, 10);
```

Comment:
We could probably make this 10 that's used here and down below as a constant like BASE_DEPENDENT_AMOUNT or something that can be reused in case it changes.

Reply:
Ahh, totally see how this looks like the array count. The 10 is actually the radix in parseInt which represents the decimal system. I just started using this recently when I found out that NaN errors can result when the value is outside of the range. check this out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt